x11-libs/wxGTK/wxGTK-3.2.1.ebuild

```diff
@@ -0,0 +1,182 @@
+
+DESCRIPTION="GTK+ version of wxWidgets, a cross-platform C++ GUI toolkit"
+HOMEPAGE="https://wxwidgets.org/"
+SRC_URI="
+	https://github.com/wxWidgets/wxWidgets/releases/download/v${PV}/wxWidgets-${PV}.tar.bz2
+	https://dev.gentoo.org/~leio/distfiles/wxGTK-3.0.5_p20210214.tar.xz
+	doc? ( https://github.com/wxWidgets/wxWidgets/releases/download/v${PV}/wxWidgets-${PV}-docs-html.tar.bz2 )"
+S="${WORKDIR}/wxWidgets-${PV}"
+
+LICENSE="wxWinLL-3 GPL-2 doc? ( wxWinFDL-3 )"
+SLOT="${WXRELEASE}"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux"
+IUSE="+X doc debug gstreamer libnotify opengl pch sdl test tiff webkit"
+REQUIRED_USE="test? ( tiff ) tiff? ( X )"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	>=app-eselect/eselect-wxwidgets-20131230
+	dev-libs/expat[${MULTILIB_USEDEP}]
+	sdl? ( media-libs/libsdl2[${MULTILIB_USEDEP}] )
+	X? (
+		>=dev-libs/glib-2.22:2[${MULTILIB_USEDEP}]
+		media-libs/libjpeg-turbo:=[${MULTILIB_USEDEP}]
+		media-libs/libpng:0=[${MULTILIB_USEDEP}]
+		sys-libs/zlib[${MULTILIB_USEDEP}]
+		x11-libs/cairo[${MULTILIB_USEDEP}]
+		x11-libs/gtk+:3[${MULTILIB_USEDEP}]
+		x11-libs/gdk-pixbuf[${MULTILIB_USEDEP}]
+		x11-libs/libSM[${MULTILIB_USEDEP}]
+		x11-libs/libX11[${MULTILIB_USEDEP}]
+		x11-libs/libXxf86vm[${MULTILIB_USEDEP}]
+		x11-libs/pango[${MULTILIB_USEDEP}]
+		gstreamer? (
+			media-libs/gstreamer:1.0[${MULTILIB_USEDEP}]
+			media-libs/gst-plugins-base:1.0[${MULTILIB_USEDEP}]
+		)
+		libnotify? ( x11-libs/libnotify[${MULTILIB_USEDEP}] )
+		opengl? ( virtual/opengl[${MULTILIB_USEDEP}] )
+		tiff? ( media-libs/tiff:0[${MULTILIB_USEDEP}] )
+		webkit? ( net-libs/webkit-gtk:4 )
+	)"
+DEPEND="${RDEPEND}
+	opengl? ( virtual/glu[${MULTILIB_USEDEP}] )
+	X? ( x11-base/xorg-proto )"
+BDEPEND="
+	test? ( >=dev-util/cppunit-1.8.0 )
+	>=app-eselect/eselect-wxwidgets-20131230
+	virtual/pkgconfig"
+
+PATCHES=(
+	#"${WORKDIR}"/wxGTK-3.0.5_p20210214/
+	"${FILESDIR}"/${P}-gtk3-translation-domain.patch
+	#"${FILESDIR}"/wxGTK-ignore-c++-abi.patch #676878
+	"${FILESDIR}/${P}-configure-tests.patch"
+)
+
+src_prepare() {
+	default
+
+	# Versionating
+	sed -i \
+		-e "s:\(WX_RELEASE = \).*:\1${WXRELEASE}:"\
+		-e "s:\(WX_RELEASE_NODOT = \).*:\1${WXRELEASE_NODOT}:"\
+		-e "s:\(WX_VERSION = \).*:\1${WXVERSION}:"\
+		-e "s:aclocal):aclocal/wxwin${WXRELEASE_NODOT}.m4):" \
+		-e "s:wxstd.mo:wxstd${WXRELEASE_NODOT}.mo:" \
+		-e "s:wxmsw.mo:wxmsw${WXRELEASE_NODOT}.mo:" \
+		Makefile.in tests/Makefile.in || die
+
+	sed -i \
+		-e "s:\(WX_RELEASE = \).*:\1${WXRELEASE}:"\
+		utils/wxrc/Makefile.in || die
+
+	sed -i \
+		-e "s:\(WX_VERSION=\).*:\1${WXVERSION}:" \
+		-e "s:\(WX_RELEASE=\).*:\1${WXRELEASE}:" \
+		-e "s:\(WX_SUBVERSION=\).*:\1${WXSUBVERSION}:" \
+		-e '/WX_VERSION_TAG=/ s:${WX_RELEASE}:3.0:' \
+		configure || die
+}
+
+multilib_src_configure() {
+	# X independent options
+	local myeconfargs=(
+		--with-zlib=sys
+		--with-expat=sys
+		--enable-compat28
```

Comment:
This looks suspect. I think we don't need such compatibility anymore, if it's even there anymore (wxWidgets doesn't use autoconf, so it might not spit a warning when it's ineffective). `compat30` would be the thing to consider now, rather than 2.8 compat.

Reply:
--enable-compat28 can even cause build errors:
https://bugs.gentoo.org/873124#c6
https://gitlab.com/kicad/code/kicad/-/issues/12665

upstream wxwidges has dropped that 2.8 compatibility code anyway in master branch:
https://github.com/wxWidgets/wxWidgets/commit/5e2da5a1066332c86f6c31223e9733337b438163
https://github.com/wxWidgets/wxWidgets/pull/22692

Reply:
@leio Thanks updated commit.